server/routes/snippet.js

```diff
@@ -8,21 +8,140 @@ const router = express.Router();
 
 /////// Upload code route handler ///////
 router.post("/upload", auth, balance, async (req, res) => {
-	const snippet = new Snippet({ ...req.body, author: req.user._id });
-	try {
-		await snippet.save();
-		req.user.balance -= 1;
-		await req.user.save();
-		res.status(201).send(snippet);
-	} catch (e) {
-		res.status(400).send(e);
-	}
+  const snippet = new Snippet({ ...req.body, author: req.user._id });
+  try {
+    await snippet.save();
+    req.user.balance -= 1;
+    await req.user.save();
+    res.status(201).send(snippet);
+  } catch (e) {
+    res.status(400).send(e);
+  }
 });
 
 /////// Retrieve code route handler ///////
 router.get("/snippet", auth, async (req, res) => {
-	const snippets = await Snippet.find({ author: req.user._id });
-	res.send(snippets);
+  const snippets = await Snippet.find({ author: req.user._id });
+  res.send(snippets);
 });
 
+// fetch requested reviews
+router.get("/requested", auth, async (req, res) => {
+  try {
+    const requestedReviews = await Snippet.find({ author: req.user._id })
+      .populate("author", ["name"])
+      .populate("reviewer", ["name"]);
+    return res.status(200).json(requestedReviews);
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ message: err.message });
+  }
+});
+
+// fetch received reviews
+router.get("/received", auth, async (req, res) => {
+  try {
+    const receivedReviews = await Snippet.find({
+      reviewer: req.user._id,
+    })
+      .populate("author", ["name"])
+      .populate("reviewer", ["name"]);
+
+    return res.status(200).json(receivedReviews);
+  } catch (error) {
+    console.error(err);
+    res.status(500).json({ message: err.message });
+  }
+});
+
+// accept a review
+router.patch("/accept/:review_id", auth, async (req, res) => {
+  try {
+    const receivedReviews = await Snippet.find({ reviewer: req.user._id });
+
+    if (!receivedReviews) return res.status(400).send("Not Found");
+
+    // find index of a specific review
+    const objIndex = receivedReviews.findIndex(
+      (obj) => obj._id.toString() === req.params.review_id
+    );
+    // change status and date_accepted
+    receivedReviews[objIndex].status = "in-review";
+    receivedReviews[objIndex].date_accepted = Date.now();
+
+    await receivedReviews[objIndex].save();
+
+    return res.status(200).json(receivedReviews);
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ message: err.message });
+  }
+});
+
+// decline a review
+router.patch("/decline/:review_id", auth, async (req, res) => {
+  try {
+    const receivedReviews = await Snippet.find({ reviewer: req.user._id });
+
```

Comment:
Similar question as the above one. And shouldn't we push this `user._id` to the `declined` field of the snippet, and send request to someone else?

Reply:
Only assigned user can decline the review